actionpack/lib/action_controller/test_case.rb

```diff
@@ -74,14 +74,21 @@ def assign_parameters(routes, controller_path, action, parameters, generated_pat
       non_path_parameters = {}
       path_parameters = {}
 
+      if parameters[:format] == :json
```

Comment:
Just realizing that we should handle `xml` request, and potentially other format ?

This test case don't seem to work 

```diff
--- a/actionpack/test/controller/test_case_test.rb
+++ b/actionpack/test/controller/test_case_test.rb
@@ -50,7 +50,8 @@ def render_body
     end

     def test_params
-      render plain: ::JSON.dump(params.to_unsafe_h)
+      render plain: params.to_unsafe_h.to_xml
     end

     def test_query_parameters
@@ -607,6 +608,12 @@ def test_using_as_json_sets_format_json
     assert_equal "json", @request.format
   end

+  def test_using_as_json_with_empty_params
+    post :test_params, params: { foo: { bar: [] } }, as: :xml
+  end
+
   def test_mutating_content_type_headers_for_plain_text_files_sets_the_header
     @request.headers["Content-Type"] = "text/plain"
     post :render_body, params: { name: "foo.txt" }
```

Reply:
Thanks, working on fixing this 👍 

Would it be helpful for me to merge this now and then send a second PR for xml?

Reply:
Sounds great 👍